thrift/thrift-gen/compile_test.go

```diff
@@ -231,13 +288,10 @@ func runBuildTest(t *testing.T, thriftFile string) error {
 }
 
 func runTest(t *testing.T, opts processOptions, extraChecks func(string) error) error {
-	tempDir, err := ioutil.TempDir("", "thrift-gen")
-	if err != nil {
-		return err
-	}
+	tempDir := getOutputDir(t)
 
 	// Generate code from the Thrift file.
-	*packagePrefix = "../"
+	*packagePrefix = strings.TrimPrefix(tempDir, _testGoPath+"/src/") + "/"
```

Comment:
Looks like package prefix will always be `github.com/uber/tchannel-go/$rand`
where `$rand` was generated by `getOutputDir`. Might be cleaner to just have
`getOutputDir` return the directory and the import path.